db/migrate/20230421202218_fix_bad_user_roles.rb

```diff
@@ -0,0 +1,10 @@
+class FixBadUserRoles < ActiveRecord::Migration[7.0]
+  def change
+    bad_roles = Role.all.select { |r| r.resource_type && r.resource.nil? }
+
+    bad_roles.each do |role|
+      UsersRole.where(role_id: role.id).destroy_all
+      role.destroy
```

Comment:
Should we do
```suggestion
      role.destroy!
```
so if something goes wrong it gets angrier?

Reply:
Hmm... no, I don't think so. We can verify the data afterwards if something failed to delete, but I wouldn't want it to hold up the rest of the migration.